### PR TITLE
GTF Sniffer Fix

### PR DESCRIPTION
**Pull request by mtekman**
Rejects a GTF file only if it lacks both "gene_id" AND "transcript_id"

### CODE REVIEW

File: lib/galaxy/datatypes/interval.py [hunk 1]
```diff
@@ -946,6 +946,9 @@ def sniff_prefix(self, file_prefix):
         >>> fname = get_test_fname('gff.gff3')
         >>> Gff3().sniff( fname )
         True
+        >>> fname = get_test_fname( 'grch37.75.gtf' )
+        >>> Gtf().sniff( fname )
```
Comment (mtekman) [inline lib/galaxy/datatypes/interval.py]:
```suggestion
        >>> Gff3().sniff( fname )
```

File: lib/galaxy/datatypes/interval.py [hunk 2]
```diff
@@ -1050,9 +1056,7 @@ def sniff_prefix(self, file_prefix):
                     # Check attributes for gene_id, transcript_id
                     attributes = parse_gff_attributes(hdr[8])
                     if len(attributes) >= 2:
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/interval.py]:
```suggestion
```

File: lib/galaxy/datatypes/interval.py [hunk 3]
```diff
@@ -1050,9 +1056,7 @@ def sniff_prefix(self, file_prefix):
...
-                        if 'gene_id' not in attributes:
-                            return False
-                        if 'transcript_id' not in attributes:
+                        if 'gene_id' not in attributes and 'transcript_id' not in attributes:
                             return False
                     else:
                         return False
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/interval.py]:
```suggestion
                    if 'gene_id' not in attributes:
                        return False
```

File: lib/galaxy/datatypes/interval.py [hunk 4]
```diff
@@ -1050,9 +1056,7 @@ def sniff_prefix(self, file_prefix):
                     # Check attributes for gene_id, transcript_id
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/interval.py]:
```suggestion
                    # Check attributes for gene_id (transcript_id is also mandatory
                    # but not for genes)
```

**Comment by mtekman** [comment]
thanks @nsoranzo !

**Review by nsoranzo — APPROVED** [review]

**Pull request merged**